src/cli/cli_coap.cpp

```diff
@@ -270,6 +330,57 @@ template <> otError Coap::Process<Cmd("stop")>(Arg aArgs[])
     return otCoapStop(GetInstancePtr());
 }
 
+/**
+ * @cli coap parameters (get,set)
+ * @code
+ * coap parameters request
+ * Transmission parameters for request:
+ * ACK_TIMEOUT=1000 ms, ACK_RANDOM_FACTOR=255/254, MAX_RETRANSMIT=2
+ * Done
+ * @endcode
+ * @code
+ * coap parameters request default
+ * Transmission parameters for request:
+ * default
+ * Done
+ * @endcode
+ * @code
+ * coap parameters request 1000 255 254 2
+ * Transmission parameters for request:
+ * ACK_TIMEOUT=1000 ms, ACK_RANDOM_FACTOR=255/254, MAX_RETRANSMIT=2
+ * Done
+ * @endcode
+ * @cparam coap parameters @ca{type} [@ca{default} | <!--
+ * -->@ca{ack_timeout ack_random_factor_numerator <!--
+ * -->ack_random_factor_denominator max_retransmit}]
+ *   * `type`: `request` for CoAP requests, or `response` for CoAP responses.
+        If no more parameters are given, the command prints the current configuration.
+ *   * `default`: Sets the transmission parameters to
+        the following default values:
+ *       * `ack_timeout`: 2000 milliseconds
+ *       * `ack_random_factor_numerator`: 3
+ *       * `ack_random_factor_denominator`: 2
+ *       * `max_retransmit`: 4
+ *   * `ack_timeout`: The `ACK_TIMEOUT` (0~UINT32_MAX) in milliseconds.
+       Refer to RFC7252.
+ *   * `ack_random_factor_numerator`:
+       The `ACK_RANDOM_FACTOR` numerator, with possible values
+       of 0-255. Refer to RFC7252.
+ *   * `ack_random_factor_denominator`:
+ *     The `ACK_RANDOM_FACTOR` denominator, with possible values
+ *     of 0-255. Refer to RFC7252.
+ *   * `max_retransmit`: The `MAX_RETRANSMIT` (0-255). Refer to RFC7252.
+ * @par
+ * Gets current CoAP parameter values if the command is run with no optional
+ * parameters.
+ * @par
+ * Sets the CoAP parameters either to their default values or to the values
+ * you specifiy, depending on the syntax chosen.
+ * @sa otCoapSendRequestWithParameters
+ * @sa otCoapSendResponseWithParameters
+ * @sa otCoapSendRequestBlockWiseWithParameters
+ * @sa otCoapSendResponseBlockWiseWithParameters
```

Comment:
For my understanding, where do you get these from?

Reply:
@Vyrastas - Hi Jeff - from scrolling through the list of CoAP functions, those four seemed applicable to refer to here.
Do you or @abtink or @jwhui want me to remove them?

Reply:
I am myself not a  fan of `@sa` in general.

Reply:
@abtink - Hi Abtin - Should I remove them?

Reply:
Let's remove them if they aren't directly reference in the command code block